test/route_end_to_end_connectivity_test.cc

```diff
@@ -0,0 +1,291 @@
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <gtest/gtest.h>
+#include <maliput/api/compare.h>
+#include <maliput/api/lane.h>
+#include <maliput/api/lane_data.h>
+#include <maliput/api/regions.h>
+#include <maliput/api/road_geometry.h>
+#include <maliput/api/road_network.h>
+#include <maliput/common/filesystem.h>
+#include <maliput/common/maliput_throw.h>
+#include <maliput/routing/phase.h>
+#include <maliput/routing/route.h>
+#include <maliput_malidrive/builder/params.h>
+#include <maliput_malidrive/builder/road_network_builder.h>
+#include <maliput_malidrive/loader/loader.h>
+
+#include "assert_compare.h"
+
+// @file This file evaluates the end to end connectivity constraints of a maliput::routing::Route.
+// @details This is not done in maliput because of the need of a concrete maliput::api::RoadGeometry
+// to evaluate all the topological and geometrical constraints. Mocking that for the purposes of
+// this test would require implementing a concrete backend. In this test, an ad-hoc xodr map is used.
+// We don't need any fancy geometry or network topology, just two roads with two lanes each are enough
+// to validate the end to end connectivity is correct.
+
+namespace maliput {
+namespace test {
+namespace {
+
+// Converts @p errors into a single string by concatenating each item with a " | " as separation.
+// @param errors A vector of strings.
+// @return A string with all items in @p error concatenated with " | " as separation.
+std::string SerializeErrors(const std::vector<std::string>& errors) {
+  std::string result;
+  for (const std::string& error : errors) {
+    result += error + " | ";
+  }
+  return result;
+}
+
+class MultipleStraightSegmentsTest : public ::testing::Test {
+ public:
+  //@{  Tolerances set to match the involved geometries and the parser resolution.
+  static constexpr double kLinearTolerance{1e-6};
+  static constexpr double kAngularTolerance{1e-6};
+  static constexpr double kScaleLength{1.0};
+  const std::string kXodrFileName{"MultipleStraightSegments.xodr"};
+  //@}
+
+  static constexpr double kLaneLength{100.};
+  const api::LaneId kLane_1_0_1{"1_0_1"};
+  const api::LaneId kLane_1_0_m1{"1_0_-1"};
+  const api::LaneId kLane_2_0_1{"2_0_1"};
+  const api::LaneId kLane_2_0_m1{"2_0_-1"};
+  const api::Lane* lane_1_0_1;
+  const api::Lane* lane_1_0_m1;
+  const api::Lane* lane_2_0_1;
+  const api::Lane* lane_2_0_m1;
+
+  std::unique_ptr<api::RoadNetwork> road_network_{};
```

Comment:
Move this to the bottom of this section.
https://google.github.io/styleguide/cppguide.html#Declaration_Order

Reply:
[5415ba7](https://github.com/maliput/maliput_integration_tests/pull/81/commits/5415ba7b4a80d037e1a4fe16999f64b6bfd528aa)